samples/Ark.ReferenceProject/Core/Ark.Reference.Core.WebInterface/Startup.cs

```diff
@@ -56,6 +57,40 @@ public override void ConfigureServices(IServiceCollection services)
         {
             base.ConfigureServices(services);
 
+            // Configure System.Text.Json source generation with Ark defaults
+            // Using JsonTypeInfoResolver.Combine to merge application and ProblemDetails contexts
+            // See: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/source-generation
+            var jsonOptions = new System.Text.Json.JsonSerializerOptions();
+            System.Text.Json.Extensions.ConfigureArkDefaults(jsonOptions);
+            
+            var coreApiContext = new CoreApiJsonSerializerContext(jsonOptions);
+            
+            // Create separate options instance for ProblemDetails context
+            var problemDetailsOptions = new System.Text.Json.JsonSerializerOptions();
+            System.Text.Json.Extensions.ConfigureArkDefaults(problemDetailsOptions);
+            var problemDetailsContext = new Ark.Tools.AspNetCore.JsonContext.ArkProblemDetailsJsonSerializerContext(problemDetailsOptions);
+            
+            // Combine source-generated contexts with minimal reflection fallback
+            // The fallback is required only for Hellang.Middleware.ProblemDetails internal types
+            // (DeveloperProblemDetailsExtensions.ErrorDetails) when IncludeExceptionDetails is enabled.
+            // This type is internal to the library and cannot be referenced in source generation.
+            // In production, exception details are typically disabled, so reflection is rarely used.
+            // The source-generated contexts handle 99%+ of serialization for optimal performance.
+            var combinedResolver = System.Text.Json.Serialization.Metadata.JsonTypeInfoResolver.Combine(
+                coreApiContext,                              // Application types - source generated (Priority 1)
+                problemDetailsContext,                        // Error types - source generated (Priority 2)
+                new System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver()); // Fallback for middleware internals (Priority 3)
+
+            services.ConfigureHttpJsonOptions(options =>
+            {
+                options.SerializerOptions.TypeInfoResolver = combinedResolver;
+            });
+
+            services.Configure<Microsoft.AspNetCore.Mvc.JsonOptions>(options =>
+            {
+                options.JsonSerializerOptions.TypeInfoResolver = combinedResolver;
```

Comment:
consider ways to ensure the JsonSerializerOptions are the ones used to initialize the Contexts / TypeInfoResolvers